packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,288 @@
+        {getName(obj)}
+      </TableData>
+      <TableData {...tableColumnInfo[1]} activeColumnIDs={activeColumnIDs}>
+        <OperandStatus operand={obj} />
+      </TableData>
+      <TableData {...tableColumnInfo[2]} activeColumnIDs={activeColumnIDs}>
+        {resource.provider}
+      </TableData>
+      <TableData {...tableColumnInfo[3]} activeColumnIDs={activeColumnIDs}>
+        {String(resource?.buckets) + " Buckets"}
+      </TableData>
+      <TableData {...tableColumnInfo[4]} activeColumnIDs={activeColumnIDs}>
+        <Kebab
+          launchModal={launchModal}
+          extraProps={{ resource: obj, resourceModel: resourceModel }}
+          customKebabItems={kebabActions}
+        />
+      </TableData>
+    </>
+  );
+};
+
+type GenericListPageProps = {
+  actions?: ModalMap;
+  resourceModel: K8sModel;
+  kebabActions?: CustomData["kebabActions"];
+};
+
+const DataResourceListView: React.FC<GenericListPageProps> = ({
+  actions,
+  resourceModel,
+  kebabActions
+}) => {
+  const { t } = useTranslation();
+  const [Modal, modalProps, launchModal] = useModalLauncher(actions);
+
+  const bucketResource = React.useMemo(
+    () => ({
+      kind: referenceForModel(NooBaaBucketClassModel),
+      namespace: DATA_FEDERATION_NAMESPACE,
+      isList: true
+    }),
+    []
+  );
+  const [bucketResources] =
+    useK8sWatchResource<BucketClassKind[]>(bucketResource);
+  let bucketClassDict = {};
+  bucketResources.forEach((bc) => {
+    if (bc.spec?.namespacePolicy?.type === BucketClassType.SINGLE) {
+      const ns = bc.spec.namespacePolicy.single.resource;
+      bucketClassDict[ns] = bucketClassDict[ns] + 1 || 1;
+    } else if (bc.spec?.namespacePolicy?.type === BucketClassType.MULTI) {
+      const dataResourceSet = new Set<string>();
+      dataResourceSet.add(bc.spec.namespacePolicy.multi.writeResource);
+      bc.spec.namespacePolicy.multi.readResources.forEach((r) =>
+        dataResourceSet.add(r)
+      );
+      dataResourceSet.forEach(
+        (ns) => (bucketClassDict[ns] = bucketClassDict[ns] + 1 || 1)
+      );
+    }
+  });
+  const resource = React.useMemo(
+    () => ({
+      kind: referenceForModel(resourceModel),
+      namespace: DATA_FEDERATION_NAMESPACE,
+      isList: true
+    }),
+    [resourceModel]
+  );
+  const [resources, loaded, loadError] =
+    useK8sWatchResource<K8sResourceCommon[]>(resource);
+  const [data, filteredData, onFilterChange] = useListPageFilter(resources);
+  const updatedDataResourceList = (filteredData as NamespaceStoreKind[]).map(
+    (a) => {
+      return {
+        ...a,
+        provider: Object.keys(NOOBAA_TYPE_MAP).find(
+          (key) => NOOBAA_TYPE_MAP[key] === a.spec.type
+        ),
+        buckets: bucketClassDict[a.metadata.name]
+          ? bucketClassDict[a.metadata.name]
+          : 0
+      };
+    }
+  );
+  const createLink = `/dfr/resource/${referenceForModel(
+    resourceModel
+  )}/create/~new`;
+
```

Comment:
```suggestion
const loaded = namespaceStoresLoaded && bucketClassesLoaded;
const loadError = namespaceStoresError || bucketClassesError;
```
then pass this `loaded` and `loadError` to all the components where it is required.